cscs-checks/tools/io/cdo.py

```diff
@@ -78,8 +78,8 @@ def __init__(self, **kwargs):
         self.descr = ('verifies compatibility with the NCO module')
         self.sourcesdir = None
         self.executable = 'echo'
-        self.sanity_patterns = sn.all([
-            sn.assert_not_found(r'.+', self.stderr)])
+        self.sanity_patterns = sn.assert_not_found(r'(ERROR)|(conflict)',
```

Comment:
I think you should use a case insensitive search using `(?i)`, as for example `r'(?i)error|conflict'`. 

The `(?i)` needs to stand only at the beginning of the pattern ( [http://www.rexegg.com/regex-disambiguation.html#modifiers](url)):
```
All popular regex flavors apart from JavaScript support inline modifiers, which allow you to tell the engine, in a pattern, to change how to interpret the pattern. For instance, (?i) turns on case-insensitivity. Except in Python, (?-i) turns it off. 

If a modifier appears at the head of the pattern, it modifies the matching mode for the whole pattern—unless it is later turned off. But (except in Python) a modifier can appear in mid-pattern, in which case in only affects the portion of the pattern that follows. 
```
I have just seen that in the `InfoNC4CCheck` you can replace `r'(?i)unsupported|(?i)error'` with `r'(?i)unsupported|error'`.

Reply:
As we try to capture here any possible error message, I think it would make sense to increase the list for keywords that indicate an error, I.e. you could maybe add `unsupported` and `failure` and other words alike to the pattern.